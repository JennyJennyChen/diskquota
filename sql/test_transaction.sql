-- Test schema
create schema ts1;
select diskquota.set_schema_quota('ts1', '1 MB');
BEGIN;
create table ts1.a(i int);
-- expect insert succeed
insert into ts1.a select generate_series(1,100);
-- expect insert fail
insert into ts1.a select generate_series(1,100000000);
END;

SELECT pg_sleep(5);

BEGIN;
create table ts1.a(i int);
-- expect insert succeed
insert into ts1.a select generate_series(1,100);
create table ts1.a2(i int);
-- expect insert succeed
insert into ts1.a2 select generate_series(1,100);
END;

-- expect insert succeed
insert into ts1.a2 select generate_series(1,100);
-- expect insert fail
insert into ts1.a select generate_series(1,100000000);

BEGIN;
-- expect insert fail
insert into ts1.a2 select generate_series(1,100);
END;


BEGIN;
drop table ts1.a;
ABORT;

select pg_sleep(5);

BEGIN;
-- expect insert fail
insert into ts1.a2 select generate_series(1,100);
END;

BEGIN;
drop table ts1.a;
END;

select pg_sleep(5);

BEGIN;
-- expect insert succeed
insert into ts1.a2 select generate_series(1,100);
END;

BEGIN;
create table ts1.a(i int);
END;

-- expect insert fail
insert into ts1.a select generate_series(1,100000000);

BEGIN;
truncate table ts1.a;
truncate table ts1.a2;
savepoint p1;
drop table ts1.a;
rollback to p1;
END;

select pg_sleep(5);

BEGIN;
-- expect insert succeed
insert into ts1.a select generate_series(1,100);
-- expect insert fail
insert into ts1.a select generate_series(1,100000000);
END;

-- will not cause error in work process and lead it to exit
BEGIN;
create table ts1.a3 (i int);
ABORT;

drop schema ts1 CASCADE;

