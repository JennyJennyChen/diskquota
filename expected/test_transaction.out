-- Test schema
create schema ts1;
select diskquota.set_schema_quota('ts1', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

BEGIN;
create table ts1.a(i int);
-- expect insert succeed
insert into ts1.a select generate_series(1,100);
-- expect insert fail
insert into ts1.a select generate_series(1,100000000);
ERROR:  schema's disk space quota exceeded with name:ts1
END;
SELECT pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

BEGIN;
create table ts1.a(i int);
-- expect insert succeed
insert into ts1.a select generate_series(1,100);
create table ts1.a2(i int);
-- expect insert succeed
insert into ts1.a2 select generate_series(1,100);
END;
-- expect insert succeed
insert into ts1.a2 select generate_series(1,100);
-- expect insert fail
insert into ts1.a select generate_series(1,100000000);
ERROR:  schema's disk space quota exceeded with name:ts1
BEGIN;
-- expect insert fail
insert into ts1.a2 select generate_series(1,100);
ERROR:  schema's disk space quota exceeded with name:ts1
END;
BEGIN;
drop table ts1.a;
ABORT;
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

BEGIN;
-- expect insert fail
insert into ts1.a2 select generate_series(1,100);
ERROR:  schema's disk space quota exceeded with name:ts1
END;
BEGIN;
drop table ts1.a;
END;
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

BEGIN;
-- expect insert succeed
insert into ts1.a2 select generate_series(1,100);
END;
BEGIN;
create table ts1.a(i int);
END;
-- expect insert fail
insert into ts1.a select generate_series(1,100000000);
ERROR:  schema's disk space quota exceeded with name:ts1
BEGIN;
truncate table ts1.a;
truncate table ts1.a2;
savepoint p1;
drop table ts1.a;
rollback to p1;
END;
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

BEGIN;
-- expect insert succeed
insert into ts1.a select generate_series(1,100);
-- expect insert fail
insert into ts1.a select generate_series(1,100000000);
ERROR:  schema's disk space quota exceeded with name:ts1
END;
-- will not cause error in work process and lead it to exit
BEGIN;
create table ts1.a3 (i int);
ABORT;
drop schema ts1 CASCADE;
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to table ts1.a2
drop cascades to table ts1.a
